doc/dart_documentation_comment_specification.md

```diff
@@ -0,0 +1,478 @@
+# **Dart Documentation Comments Specification**
+Version: 1.0.0
+
+[TOC]
+
+## **1\. Introduction**
+
+### **1.1. Purpose**
+
+The purpose of this document is to provide a single specification for documentation comments in the Dart programming language. Its goal is to ensure that documentation is parsed consistently by all tools and to serve as a definitive standard for developers and tool authors.
+
+### **1.2. Scope**
+
+**In Scope**
+
+This specification covers the following topics:
+
+* The *syntax* for writing documentation comments.
+* The rules governing where documentation comments can be *placed*.
+* The mechanism for *resolving* references between elements.
+* The logic for *associating* documentation with code elements.
+
+**Out of Scope**
+
+This document does not cover:
+
+* The implementation details of specific documentation tools (e.g., `dartdoc`, `analysis server`).
+* The syntax and behavior of tool-specific directives (e.g., `@template`, `@canonicalFor`).
+* Any details related to the visual styling or HTML layout of generated documentation.
+* Best practices or style guides for writing effective documentation content.
+
+### **1.3. Terminology**
+
+* **Doc Comment:** A comment intended to be processed by documentation generation tools, like  dartdoc.
+* **Element**: A specific, declared element in the Dart code, such as a class, method, function, variable, or type parameter.
```

Comment:
(Cyclic definition: An element is a (specific declared) element. Then some examples, but one has to guess what they have in common.
The concept of "element" is an implementation detail in our tools, it's not a term that the language specification uses. It's not a word we can assume a reader will understand by themselves, so if we want to use it here, it needs to be defined, non-circularly. Or just use "declaration" or maybe "named declaration".)